### ISSUE TITLE
Don't show warning with `static inline` functions in headers

### ISSUE THREAD

**Issue by 8dcc** [flycheck/flycheck#2112]
## Checklist

- [X] I have searched existing issues and StackOverflow for solutions.
- [X] I have read the [Flycheck manual][].
- [X] I have read the [Troubleshooting Guide][].
- [X] I have checked my issue has not been reported yet.

## Problem description

After enabling `global-flycheck-mode`, when opening a C header file (using `c-mode`) which contains a function declared as `static inline`, a warning pops up because the function is not being used. This makes sense in a source file, but it's not strange to see them defined in headers, since they are included in other sources.

Is there a way of disabling/configuring this specific warning?

It's my first time trying flycheck (or any syntax checker like that in Emacs), so I apologize if I did something wrong.

## Steps to reproduce

1. Open `/tmp/a.h` and enable `c-mode`.
2. Write `static inline int f(int x) { return x; }`.
3. See warning.

- [X] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior

Don't show this warning on header files.

## System configuration

```
Syntax checkers for buffer a.h in c-mode:

First checker to run:

  c/c++-clang
    - may enable:    yes
    - executable:    Found at /usr/bin/clang
    - next checkers: c/c++-cppcheck

Checkers that could run if selected:

  c/c++-gcc  select
    - may enable:    yes
    - executable:    Found at /usr/bin/gcc
    - next checkers: c/c++-cppcheck

Checkers that are compatible with this mode, but will not run until properly configured:

  c/c++-cppcheck (automatically disabled) reset
    - may enable: no
    - executable: Not found

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 35.0-snapshot
Emacs version:    29.4
System:           x86_64-pc-linux-gnu
Window system:    x
```

Emacs configuration:
- [X] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Prelude
- [ ] Other shared configuration (please specify)

[Flycheck manual]:  https://www.flycheck.org/en/latest/user/troubleshooting.html
[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html
